lib/index.js

```diff
@@ -888,6 +894,312 @@ module.exports = () => {
         throw new Error('ec_seckey_tweak_mul', ret);
       }
 
+      /**
+       * @summary check if a point is valid on the curve
+       * @argumentument {Uint8Array} point - point to check
+       * @returns {boolean} true if valid, false otherwise
+       */
+      function isPoint(point) {
+        if (!point || !(point instanceof Uint8Array)) {
+          throw new TypeError('point must be a Uint8Array');
+        }
+
+        const pointPtr = charStar(point);
+
+        if (point.length === 32) {
+          const validXonlyPoint = Module.ccall(
+            'ec_is_valid_xonly_pubkey',
+            'number',
+            ['number'],
+            [pointPtr]
+          );
+
+          freeMalloc();
+          return validXonlyPoint === 1;
+        } else {
+          const validEcPubkey = Module.ccall(
+            'ec_is_valid_ec_pubkey',
+            'number',
+            ['number', 'number'],
+            [pointPtr, point.length]
+          );
+          freeMalloc();
+          return validEcPubkey === 1;
+        }
+      }
+
+      /**
+       *  @summary compress a valid point
+       *  @argumentument {Uint8Array} point - point to compress
+       *  @argumentument {boolean | undefined} compressed - true if compressed, false otherwise
+       *  @returns {Uint8Array} compressed point
+       **/
+      function pointCompress(point, compressed) {
+        if (!point || !(point instanceof Uint8Array)) {
+          throw new TypeError('point must be a Uint8Array');
+        }
+        if (compressed !== undefined && typeof compressed !== 'boolean') {
+          throw new TypeError('compressed must be a boolean');
+        }
+        const outputlen =
+          compressed === undefined ? point.length : compressed ? 33 : 65;
```

Comment:
this is not the right way since the content of this var will be set by the c function:
```suggestion
        const outputlen = malloc(8);
        Module.setValue(compressed? 33 : 65, 'int64');
```